src/fileviewer.cpp

```diff
@@ -392,14 +394,41 @@ void FileViewer::ShowError(const char *icon, const wxString& msg, const wxString
 
 void FileViewer::OnChoice(wxCommandEvent &event)
 {
-    SelectReference(m_references[event.GetSelection()]);
+    m_chosenRefIndex = event.GetSelection();
+    SelectReference(m_references[m_chosenRefIndex]);
 }
 
-void FileViewer::OnEditFile(wxCommandEvent&)
+void FileViewer::OnEditFile(wxCommandEvent &)
 {
     wxFileName filename = GetFilename(bidi::strip_control_chars(m_file->GetStringSelection()));
-    if (filename.IsOk())
-        wxLaunchDefaultApplication(filename.GetFullPath());
+    if (filename.IsOk()) {
+        wxFileType *c_type;
+        c_type = wxTheMimeTypesManager->GetFileTypeFromExtension(filename.GetExt());
+        if (c_type) {
+    	      wxString command, filePath;
+    	      filePath = filename.GetFullPath();
+    	      wxFileType::MessageParameters param;
+    	      wxFileType::MessageParameters &paramRef = param;
+    	      c_type->GetOpenCommand(&command, paramRef);
+
+            if ( command.EndsWith("vim ") || command.EndsWith("view ") ) {
+                command.Append("-g ");  //  To make an indepedent window instance of Vi-like editor/viewer
+
+                wxString refItem = m_references[m_chosenRefIndex];
+
+                wxString lineNumberStr = refItem.AfterLast(':');
+                if (lineNumberStr.length() < refItem.length() ) {  //  succeed to extract the line number
```

Comment:
Why don't you use `!lineNumberStr.empty()`?

Reply:
Oh, certainly it is better. I use it.